File: two-sum-sorting.py
def two_sum(nums, k):
  nums.sort()
  lp = 0
  rp = len(nums) - 1
  while lp < rp:
    pair_sum = nums[lp] + nums[rp]
    if pair_sum == k:
      return True
    elif pair_sum < k:
      lp += 1
    else:
      rp -= 1
  return False
  

File: two-sum-sorting.js
function twoSum(nums, k) {
  nums.sort();
  let lp = 0;
  let rp = nums.length - 1;
  while (lp < rp) {
    const sum = nums[lp] + nums[rp];
    if (sum === k) return true;
    else if (sum < k) lp += 1;
    else rp -= 1;
  }
  return false;
}
